src/OpenIddict.Core/Infrastructure/OpenIddictProvider.Serialization.cs

```diff
@@ -17,14 +17,41 @@
 namespace OpenIddict.Infrastructure {
     public partial class OpenIddictProvider<TUser, TApplication, TAuthorization, TScope, TToken> : OpenIdConnectServerProvider
         where TUser : class where TApplication : class where TAuthorization : class where TScope : class where TToken : class {
-        public override async Task SerializeRefreshToken([NotNull] SerializeRefreshTokenContext context) {
+        public override async Task SerializeAuthorizationCode([NotNull] SerializeAuthorizationCodeContext context) {
             var services = context.HttpContext.RequestServices.GetRequiredService<OpenIddictServices<TUser, TApplication, TAuthorization, TScope, TToken>>();
 
-            Debug.Assert(context.Request.RequestType == OpenIdConnectRequestType.TokenRequest,
-                "The request should be a token request.");
+            Debug.Assert(context.Request.RequestType == OpenIdConnectRequestType.Authentication, "The request should be an authorization request.");
+            Debug.Assert(!string.IsNullOrEmpty(context.Request.ClientId), "The client_id parameter shouldn't be null or empty.");
+
+            // Note: a null value could be returned by FindByIdAsync. In this case, throw an exception to abort the token request.
+            var user = await services.Users.FindByIdAsync(context.Ticket.Principal.GetClaim(ClaimTypes.NameIdentifier));
+            if (user == null) {
+                throw new InvalidOperationException("The token request was aborted because the user associated " +
```

Comment:
Well, we could. But seeing this exception should be extremely rare, as this method is never called if the user entity cannot be validated when receiving the authorization request. So it's only thrown if the user is removed after the initial check (which is basically a race condition).